hydra/_internal/defaults_list.py

```diff
@@ -365,20 +366,33 @@ def _update_overrides(
             assert d.group is not None
             legacy_hydra_override = not d.is_override() and d.group.startswith("hydra/")
 
-        if seen_override and not (
+        if (seen_override or seen_legacy_override) and not (
             d.is_override() or d.is_external_append() or legacy_hydra_override
         ):
             assert isinstance(last_override_seen, GroupDefault)
             pcp = parent.get_config_path()
             okey = last_override_seen.get_override_key()
             oval = last_override_seen.get_name()
-            raise ConfigCompositionException(
-                dedent(
-                    f"""\
-                    In {pcp}: Override '{okey} : {oval}' is defined before '{d.get_override_key()}: {d.get_name()}'.
-                    Overrides must be at the end of the defaults list"""
+            dkey = d.get_override_key()
+            dval = d.get_name()
```

Comment:
I added these two lines to work around linter requirements (line length <= 120 characters).